sdkproject/Assets/Mapbox/Unity/Editor/PropertyDrawers/PrefabItemOptionsDrawer.cs

```diff
@@ -0,0 +1,186 @@
+		public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+		{
+			EditorGUI.BeginProperty(position, label, property);
+
+			position.y += 1.2f*_lineHeight;
+
+			var prefabItemCoreOptions = property.FindPropertyRelative("coreOptions");
+			GUILayout.Label(prefabItemCoreOptions.FindPropertyRelative("sublayerName").stringValue + " Properties");
+			GUILayout.Space(2.5f*EditorGUIUtility.singleLineHeight);
+
+			//Prefab Game Object
+			position.y += _lineHeight;
+			var spawnPrefabOptions = property.FindPropertyRelative("spawnPrefabOptions");
+			EditorGUI.PropertyField(new Rect(position.x, position.y + 2,position.width,2*_lineHeight),spawnPrefabOptions);
+			GUILayout.Space(1);
+
+			//Prefab Locations title
+			GUILayout.Label(prefabLocationsTitle);
+
+			//FindBy drop down
+			EditorGUILayout.BeginHorizontal();
+			EditorGUILayout.PrefixLabel(findByDropDown);
+			var findByProp = property.FindPropertyRelative("findByType");
+			GUILayout.Space(shiftLeftPixels);
+			findByProp.enumValueIndex = EditorGUILayout.Popup(findByProp.enumValueIndex, findByProp.enumDisplayNames);
+			EditorGUILayout.EndHorizontal();
+
+			switch((LocationPrefabFindBy)findByProp.enumValueIndex)
+			{
+				case (LocationPrefabFindBy.MapboxCategory):
+					ShowCategoryOptions(property);
+					break;
+				case(LocationPrefabFindBy.AddressOrLatLon):
+					ShowAddressOrLatLonUI(property);
+					break;
+				case (LocationPrefabFindBy.POIName):
+					ShowPOINames(property);
+					break;
+				default:
+					break;
+			}
+
+			EditorGUI.EndProperty();
+		}
+
+		private void ShowCategoryOptions(SerializedProperty property)
+		{
+			//Category drop down
+			EditorGUILayout.BeginHorizontal();
+			EditorGUILayout.PrefixLabel(categoriesDropDown);
+
+			var categoryProp = property.FindPropertyRelative("categories");
+			GUILayout.Space(shiftLeftPixels);
+			categoryProp.intValue = (int)(LocationPrefabCategories)(EditorGUILayout.EnumFlagsField((LocationPrefabCategories)categoryProp.intValue));
+			EditorGUILayout.EndHorizontal();
+
+			ShowDensitySlider(property);
+		}
+
+		private void ShowAddressOrLatLonUI(SerializedProperty property)
+		{
+
+
+			EditorGUILayout.BeginVertical();
+			var coordinateProperties = property.FindPropertyRelative("coordinates");
+
+			for (int i = 0; i < coordinateProperties.arraySize; i++)
+			{
+				EditorGUILayout.BeginHorizontal();
+				//get the element to draw
+				var coordinate = coordinateProperties.GetArrayElementAtIndex(i);
+
+				//TODO: update coordinate property drawer from specific-locations-modifiers branch
+				//use the tagged property drawer for the coordinate layout
+				EditorGUILayout.PropertyField(coordinate);
+
+				//include a remove button in the row
+				if (GUILayout.Button(new GUIContent(" X "), (GUIStyle)"minibuttonright", GUILayout.Width(30)))
+				{
+					coordinateProperties.DeleteArrayElementAtIndex(i);
+				}
+				EditorGUILayout.EndHorizontal();
+			}
+
+			EditorGUILayout.BeginHorizontal();
```

Comment:
A lot of the GUI layout code seems a little messy/opaque (unnecessary newlines, num values that seem arbitrary, etc). It's ultimately a purely cosmetic change, but is there a plan at some point to comb through the code and clean it up?

Reply:
I like the sound of that